great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -828,6 +830,10 @@ class SQLDatasource(Datasource):
     # left side enforces the names on instance creation
     type: Literal["sql"] = "sql"
     connection_string: Union[ConfigStr, str]
+    connect_args: Dict[str, Union[ConfigStr, Any]] = pydantic.Field(
+        default={},
+        description="Optional dictionary of `connect_args` will be passed to the SQLAlchemy Engine",
+    )
```

Comment:
Should we add a validator that checks these values and raises a warning if anything is not serializable?